### PR TITLE
Add ability for confmap.Provider to be notified of config changes

### PR DESCRIPTION
**Pull request by tigrannajaryan**
This is a draft proposal for issue https://github.com/open-telemetry/opentelemetry-collector/issues/6161 that shows how potentially an OpAMP config provider can be implemented.

This an incomplete draft that shows the idea. Don't merge.

### CODE REVIEW

**Comment by tigrannajaryan** [comment]
@amdprophet @portertech take a look at this draft that partially addresses https://github.com/open-telemetry/opentelemetry-collector/issues/6161

**Comment by tigrannajaryan** [comment]
@open-telemetry/collector-approvers I am curious what you think about this approach and whether you have alternate proposals.

**Review by bogdandrutu — COMMENTED** [review]
The `confmap.Providers` are designed for only one flow of the data, and this PR changes that which is not necessary bad, but feels a bit too much in terms of capabilities.

Opamp provider will not need capabilities of merging, envars, etc. correct? If that is the case, we should probably implement the ConfigProvider interface not the confmap.Provider. What do you think?

File: service/config_provider.go [hunk 1]
```diff
@@ -60,10 +60,27 @@ type ConfigProvider interface {
...
 	// Should never be called concurrently with itself or Get.
 	Shutdown(ctx context.Context) error
+
+	// ConfigUpdateFailed is called by the Service if applying the last config
+	// returned by Get() failed for any reason, including if Get() returned an error.
+	ConfigUpdateFailed(err error)
+
+	// ConfigUpdateSucceeded is called by the Service if applying the last config
+	// return by Get() succeeded and the Service.Start() succeeded.
+	ConfigUpdateSucceeded()
```
[the remark this answers is not in the document]
Comment (tigrannajaryan) [reply service/config_provider.go:70]: Yes, I was also thinking about it as an alternate.

**Comment by tigrannajaryan** [comment]
Related to and probably blocked by https://github.com/open-telemetry/opentelemetry-collector/issues/6226

**Pull request closed without merging**